src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+            // }
+            // else if (cursorAboveTopDist > MinAutoScrollDist)
+            // {
+            //     newAutoScrollVelocity = -1.0 * _getAutoScrollSpeed(cursorAboveTopDist);
+            // }
+
+            // if (newAutoScrollVelocity != 0)
+            // {
+            //     _tryStartAutoScroll(point, newAutoScrollVelocity);
+            // }
+            // else
+            // {
+            //     _tryStopAutoScroll(ptr.PointerId());
+            // }
+        }
+
+        _core->UpdateHoveredCell(terminalPosition);
+    }
+
+    void ControlInteractivity::TouchMoved(const winrt::Windows::Foundation::Point newTouchPoint,
+                                          const bool focused)
+    {
+        if (focused &&
+            _touchAnchor)
+        {
+            const auto anchor = _touchAnchor.value();
+
+            // Our actualFont's size is in pixels, convert to DIPs, which the
+            // rest of the Points here are in.
+            const auto fontSizeInDips{ _core->FontSizeInDips() };
+
+            // Get the difference between the point we've dragged to and the start of the touch.
+            const float dy = newTouchPoint.Y - anchor.Y;
+
+            // Start viewport scroll after we've moved more than a half row of text
+            if (std::abs(dy) > (fontSizeInDips.height<float>() / 2.0f))
+            {
+                // Multiply by -1, because moving the touch point down will
+                // create a positive delta, but we want the viewport to move up,
+                // so we'll need a negative scroll amount (and the inverse for
+                // panning down)
+                const float numRows = -1.0f * (dy / fontSizeInDips.height<float>());
+
+                const auto currentOffset = ::base::ClampedNumeric<double>(_core->ScrollOffset());
+                const auto newValue = numRows + currentOffset;
+
+                // Update the Core's viewport position, and raise a
+                // ScrollPositionChanged event to update the scrollbar
+                _updateScrollbar(newValue);
+
+                // Use this point as our new scroll anchor.
+                _touchAnchor = newTouchPoint;
+            }
+        }
+    }
+
+    void ControlInteractivity::PointerReleased(TerminalInput::MouseButtonState buttonState,
+                                               const unsigned int pointerUpdateKind,
+                                               const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                               const bool /*focused*/,
+                                               const til::point terminalPosition)
+    {
+        // Short-circuit isReadOnly check to avoid warning dialog
+        if (!_core->IsInReadOnlyMode() && _canSendVTMouseInput(modifiers))
+        {
+            _trySendMouseEvent(pointerUpdateKind, buttonState, modifiers, terminalPosition);
+            return;

```

Comment:
This may not be necessary yet, but I just noticed that `_PointerReleasedHandler` in `main` updates `args.Handled(true)` when we actually do something. Should all of these functions return a `bool` so that we can choose when an event was handled and when it should be handed over to another UI element? That sounds like it might be useful, and not doing so may be a source of bugs?

Reply:
So, looking over how they are in main vs now, it looks like they largely all mark `Handled` as `true`, or if they didn't, that logic still lives in `TermControl`. I'm not opposed to the idea, it just seems like it's all `return true` for now.